src/components/Filters/SchemaSieve.ts

```diff
@@ -1,157 +1,173 @@
-				if (!!(<AdvancedPineOperatorTest>value).getLabel) {
-					const label = (<AdvancedPineOperatorTest>value).getLabel(schemaEntry);
-					if (label) {
-						return {
-							label,
-							value: key,
-						};
-					}
-				}
-				return {
-					label: key,
-					value: key,
-				};
-			});
+	if (type === 'string') {
+		if (format === 'date-time') {
+			return DataTypes.date_time;
 		}
-		return [];
+		return DataTypes.string;
+	}
+	if (type === 'object') {
+		return DataTypes.object;
+	}
+	if (type === 'boolean') {
+		return DataTypes.boolean;
+	}
+	if (type === 'number') {
+		return DataTypes.number;
 	}
 
-	makeFilterInputs(schema: Schema) {
-		const inputs: {
-			[key: string]: FilterModel;
-		} = {};
-
-		forEach(schema, (value, key) => {
-			inputs[key] = {
-				type: value.type,
-				name: key,
-				label: value.label,
-				availableOperators: this.getOperators(value.type, schema[key]),
-				operator: null,
-				value: null,
-			};
-		});
-
-		return inputs;
+	return null;
+};
+
+export const createFullTextSearchFilter = (
+	schema: JSONSchema6,
+	term: string,
+) => {
+	// A schema that matches applies the pattern to each known schema field
+	const filter = {
+		$id: utils.randomString(),
+		title: FULL_TEXT_SLUG,
+		anyOf: [
+			{
+				description: `Full text search is ${term}`,
+				anyOf: Object.keys(schema.properties!).map(key => ({
+					properties: {
+						[key]: {
+							type: 'string',
+							regexp: {
+								pattern: utils.regexEscape(term),
+								flags: 'i',
+							},
+						},
+					},
+					required: [key],
+				})),
+			},
+		],
+	};
+
+	return filter as JSONSchema6;
+};
+
+// Update or insert a full text search filter into an array of filters
+export const upsertFullTextSearch = (
+	schema: JSONSchema6,
+	filters: JSONSchema6[] = [],
+	term: string,
+) => {
+	const searchFilter = createFullTextSearchFilter(schema, term);
+	const existingSearchIndex = findIndex(filters, { title: FULL_TEXT_SLUG });
+
+	if (existingSearchIndex > -1) {
+		filters.splice(existingSearchIndex, 1, searchFilter);
+	} else {
+		filters.push(searchFilter as JSONSchema6);
 	}
 
-	validate(type: string, value: any) {
-		// If the type is unknown just return true
-		if (!(type in filterTests)) {
-			return true;
+	return filters;
+};
+
+// Removes a full text search filter from an array of filters
+export const removeFullTextSearch = (filters: JSONSchema6[]) =>
+	filters.filter(f => f.title !== FULL_TEXT_SLUG);
+
+export const createFilter = (
+	schema: JSONSchema6,
+	signatures: FilterSignature[],
+): JSONSchema6 => {
+	const anyOf = signatures.map(({ field, operator, value }) => {
+		const subSchema = schema.properties![field] as JSONSchema6;
+		const model = getDataModel(subSchema);
+
+		if (!model || !model.operators.hasOwnProperty(operator)) {
+			return {};
 		}
 
-		return filterTests[type].validate(value);
-	}
+		return (model.createFilter as (...args: any[]) => JSONSchema6)(
```

Comment:
Could we define a base signature to cast `createFilter`?
```
(
field: string,
operator: string,
value: any,
subSchema: JSONSchema6,
) => JSONSchema6
```
This way we can reduce the amounts of `any` used and have a place that actually checks that all `createFilter` implementations are compatible.